ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java

```diff
@@ -176,7 +176,8 @@ protected void setState(TaskState state) {
 
   protected Hive getHive() {
     try {
-      return Hive.getWithFastCheck(conf);
+      //Hive.getWithFastCheck(conf) will cause leaked tcp link to metastore
```

Comment:
Yes,the result of isCompatible is different. the code:
  private static boolean isCompatible(Hive db, HiveConf c, boolean isFastCheck) {
        return isFastCheck? db.metaStoreClient.isSameConfObj(c) : db.metaStoreClient.isCompatibleWith(c);
  }
Because the Task and the Hive has different HiveConf. So, isCompatible will resturn false and a new Hive will be created, while the old Hive will be closed. The metaStoreClient will be closed too.
But when the txnManager releaseLocks, an exception will be thrown. Like this:

`
2017-03-01T10:35:38,753 WARN  [HiveServer2-Handler-Pool: Thread-31]: hive.metastore.RetryingMetaStoreClient (:()) - MetaStoreClient lost connection. Attempting to reconnect.org.apache.thrift.transport.TTransportException: Cannot write to null outputStream at org.apache.thrift.transport.TIOStreamTransport.write(TIOStreamTransport.java:142)
`

Because txnManager hold the old Hive's metaStoreClient which is closed. 
So the metaStoreClient will be reconnect, and a new TSocket will be created. But this metaStoreClient will not be used again untill GC happening.